scrapy/utils/signal.py

```diff
@@ -1,123 +1,39 @@
+        )
+        from .signal_ops import send_catch_log
 
+        return send_catch_log(*args, **kwargs)
 
-def send_catch_log_deferred(
-    signal: TypingAny = Any,
-    sender: TypingAny = Anonymous,
-    *arguments: TypingAny,
-    **named: TypingAny,
-) -> Deferred[List[Tuple[TypingAny, TypingAny]]]:
-    """Like send_catch_log but supports returning deferreds on signal handlers.
-    Returns a deferred that gets fired once all signal handlers deferreds were
-    fired.
-    """
+    def send_catch_log_deferred(self, *args, **kwargs):
+        warnings.warn(
+            "The `scrapy.utils.signal` module is deprecated and will be removed in a future release. "
+            "Please use `scrapy.utils.signal_ops` instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        from .signal_ops import send_catch_log_deferred
 
-    def logerror(failure: Failure, recv: Any) -> Failure:
-        if dont_log is None or not isinstance(failure.value, dont_log):
-            logger.error(
-                "Error caught on signal handler: %(receiver)s",
-                {"receiver": recv},
-                exc_info=failure_to_exc_info(failure),
-                extra={"spider": spider},
-            )
-        return failure
+        return send_catch_log_deferred(*args, **kwargs)
 
-    dont_log = named.pop("dont_log", None)
-    spider = named.get("spider", None)
-    dfds: List[Deferred[Tuple[TypingAny, TypingAny]]] = []
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        d: Deferred[TypingAny] = maybeDeferred_coro(
-            robustApply, receiver, signal=signal, sender=sender, *arguments, **named
+    def disconnect_all(self, *args, **kwargs):
+        warnings.warn(
+            "The `scrapy.utils.signal` module is deprecated and will be removed in a future release. "
+            "Please use `scrapy.utils.signal_ops` instead.",
+            DeprecationWarning,
+            stacklevel=2,
         )
-        d.addErrback(logerror, receiver)
-        # TODO https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/cell-var-from-loop.html
-        d2: Deferred[Tuple[TypingAny, TypingAny]] = d.addBoth(
-            lambda result: (
-                receiver,  # pylint: disable=cell-var-from-loop  # noqa: B023
-                result,
-            )
-        )
-        dfds.append(d2)
-    dl = DeferredList(dfds)
-    d3: Deferred[List[Tuple[TypingAny, TypingAny]]] = dl.addCallback(
-        lambda out: [x[1] for x in out]
-    )
-    return d3
+        from .signal_ops import disconnect_all
+
+        return disconnect_all(*args, **kwargs)
 
 
-def disconnect_all(signal: TypingAny = Any, sender: TypingAny = Any) -> None:
-    """Disconnect all signal handlers. Useful for cleaning up after running
-    tests
-    """
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        disconnect(receiver, signal=signal, sender=sender)
+signal = SignalDeprecated()
```

Comment:
This would be `scrapy.utils.signal.signal`

Reply:
Is my latest change to `signal.py` acceptable for maintaining backwards compatibility?

I noticed a test failure in a specific environment. Do you have any recommendations for fixing it? I was thinking we might need to `import signal as builtin_signal` in `test_crawler.py`. Alternatively, would increasing the delay at line 909 help? Or perhaps both?

Reply:
Hi @wRAR @Gallaecio , I noticed that the test that previously failed has now passed after being rerun. Just checking in to see if there's anything else you'd like me to address?